### PR TITLE
IA-3995: enable retries in http4s client

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/IA-3995

I think we inadvertently weren't enabling retries in the http4s client which we commonly use for WSM, Sam, and others. Theory is this made us un-resilient to socket exceptions, etc.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala [hunk 1]
```diff
@@ -736,7 +736,7 @@ object Boot extends IOApp {
             meteredClient = Metrics[F](
               metricsOps,
               classifierFunc
-            )(httpClientWithLogging)
+            )(clientWithRetry)
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala:739]: This code path previously wasn't using the retry-enabled client

Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala:739]: ohhh good catch 👀

**Review by LizBaldo — APPROVED** [review]

**Comment by rtitle** [comment]
BEE testing looks good, I think -- at least, I was able to create/delete a VM a few times. I also validated that all cloud resources (VM, 2 disks, nic) were all deleted after Leo runtime deletion.

**Review by aherbst-broad — APPROVED** [review]

**Comment by rtitle** [comment]
wow swatomation passed

**Pull request merged**